src/rust/collections/pin_slab.rs

```diff
@@ -114,6 +120,25 @@ impl<T> PinSlab<T> {
         }
     }
 
+    /// Get a reference to the value at the given slot.
+    #[inline(always)]
+    unsafe fn internal_get(&self, key: usize) -> Option<&T> {
```

Comment:
Why is this function unsafe? Is it possible to reduce the unsafe scope? Could you add a note on why it is safe to call the unsafe scope?

Reply:
I have limited the unsafe block to the only small bit of unsafe code. For symmetry, I have also updated internal_get_mut.